### PR TITLE
Improve visitor model and specs

### PR DESCRIPTION
**Pull request by threedaymonk**
This makes various improvements to the way that the visitor model and validations work and are tested, and corrects age calculation (as this was flagged by the new tests). It is intended to lay the groundwork for #276.

### CODE REVIEW

**Comment by threedaymonk** [comment]
Tested on preprod.

**Comment by SteveMarshall** [comment]
LGTM!

File: app/services/age_calculator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AgeCalculator
+  def age(date_of_birth, today)
+    # See https://stackoverflow.com/a/2357790
+    if today.month > date_of_birth.month ||
```
Comment (dan-palmer) [inline app/services/age_calculator.rb]: How about hoisting this guy into a predicate method, something like `had_bithday_this_year?`

File: spec/features/datepicker_spec.rb [hunk 2]
```diff
@@ -15,9 +15,6 @@
 
   before :each do
     allow_any_instance_of(VisitController).to receive(:metrics_logger).and_return(MockMetricsLogger.new)
-    allow_any_instance_of(EmailValidator).to receive(:validate_dns_records)
```
Comment (dan-palmer) [inline spec/features/datepicker_spec.rb]: :door: :walking:  :sunglasses: nice one!

File: app/services/age_calculator.rb (hunk 1 above)
Comment (threedaymonk) [inline app/services/age_calculator.rb]: Good call.

**Comment by threedaymonk** [comment]
Tested on preprod (as at ff0823c).

**Comment by dan-palmer** [comment]
LGTM! 😎

**Pull request merged**